jujulint/model_input.py

```diff
@@ -244,7 +244,13 @@ def filter_by_relation(self, apps: Set, endpoint: str) -> Set:
         apps_related = set()
         for app in apps:
             relations = self.applications_data.get(app, {}).get("relations", {})
-            apps_related.update(relations.get(endpoint, []))
+            endpoints = relations.get(endpoint, [])
+            # since juju 3.4, this is a dict instead of a string,
+            # so convert to a string.
+            for i in range(len(endpoints)):
+                if isinstance(endpoints[i], dict):
+                    endpoints[i] = endpoints[i]["related-application"]
```

Comment:
nit: avoid using `i`, `j` or other meaning less variables
```suggestion
            for endpoint in range(len(endpoints)):
                if isinstance(endpoints[endpoint], dict):
                    endpoints[endpoint] = endpoints[endpoint]["related-application"]
```

Reply:
`i` is not meaningless: it's a standard convention for an indexing integer variable.  `endpoint` in this case is misleading, because it's not an endpoint - it's an integer used for indexing the endpoints variable.

Reply:
+1 for "i" in this case.